app/src/main/java/org/secfirst/umbrella/whitelabel/feature/segment/view/SegmentController.kt

```diff
@@ -124,11 +135,77 @@ class SegmentController(bundle: Bundle) : BaseController(bundle), SegmentView {
     private fun onSegmentShareClick(markdown: Markdown) {
         val andDown = AndDown()
         val result = andDown.markdownToHtml(markdown.text, AndDown.HOEDOWN_EXT_QUOTE, 0)
-        saveHtmlFile(result)
-        val intent = Intent(Intent.ACTION_SENDTO,
-                Uri.parse("mailto:?subject=${markdown.title}&body=${Uri.encode(markdown.text)}"))
+        val doc = Jsoup.parse(result)
+
+        val img: Elements = doc.getElementsByTag("img")
+        lateinit var src: String
+        lateinit var base64img: String
+
+
+
+        for (el in img) {
+            src = el.absUrl("src").replace("file://", "")
+            base64img = encodeToBase64(File(src))
+            src = "data:image/png;base64," + base64img
+            el.attr("src", src)
+
+        }
+
+        val filename = markdown.title
+        val f = java.io.File(UmbrellaApplication.instance.cacheDir, SHARE_FILE)
```

Comment:
If you need to use the application context on View, I suggest you use context instead of `UmbrellaApplication.instance` .

For example:
~~`val f = java.io.File(UmbrellaApplication.instance.cacheDir, SHARE_FILE)`~~
`val f = java.io.File(context.cacheDir, SHARE_FILE)`